lib/_http_client.js

```diff
@@ -134,7 +135,8 @@ function ClientRequest(options, cb) {
     self._last = true;
     self.shouldKeepAlive = false;
     const optionsPath = {
-      path: self.socketPath
+      path: self.socketPath,
+      timeout: options.timeout
```

Comment:
Minor nit, but to be consistent we could just reference `self.timeout` here instead.